src/opencl/utils.rs

```diff
@@ -58,52 +55,44 @@ fn build_device_list() -> Vec<Device> {
     let platforms: Vec<_> = opencl3::platform::get_platforms().unwrap_or_default();
 
     for platform in platforms.iter() {
-        let platform_name = match platform.name() {
-            Ok(pn) => pn,
-            Err(error) => {
-                warn!("Cannot get platform name: {:?}", error);
-                continue;
-            }
-        };
-        if let Some(brand) = Brand::by_name(&platform_name) {
-            let devices = platform
-                .get_devices(opencl3::device::CL_DEVICE_TYPE_GPU)
-                .map_err(Into::into)
-                .and_then(|devices| {
-                    devices
-                        .into_iter()
-                        .map(opencl3::device::Device::new)
-                        .filter(|d| {
-                            if let Ok(vendor) = d.vendor() {
-                                match vendor.as_str() {
-                                    // Only use devices from the accepted vendors ...
-                                    AMD_DEVICE_VENDOR_STRING | NVIDIA_DEVICE_VENDOR_STRING => {
-                                        // ... which are available.
-                                        return d.available().unwrap_or(0) != 0;
-                                    }
-                                    _ => (),
-                                }
+        let devices = platform
+            .get_devices(opencl3::device::CL_DEVICE_TYPE_GPU)
+            .map_err(Into::into)
+            .and_then(|devices| {
+                devices
+                    .into_iter()
+                    .map(opencl3::device::Device::new)
+                    .filter(|d| {
+                        if let Ok(vendor) = d.vendor() {
+                            // Only use devices from the accepted vendors ...
+                            if Vendor::try_from(vendor.as_str()).is_ok() {
+                                // ... which are available.
+                                return d.available().unwrap_or(0) != 0;
                             }
-                            false
+                        }
+                        false
+                    })
+                    .map(|d| -> GPUResult<_> {
```

Comment:
could use `filter_map` instead

Reply:
I've opened https://github.com/filecoin-project/rust-gpu-tools/issues/42 for that.

Reply:
`filter_map` is implemented in https://github.com/filecoin-project/rust-gpu-tools/pull/39/files#diff-799aad7ae6bfd4ae76ce98c354a508494667af58cb53f46cd870be729603c9e6R84 as i needed it for other reasons anyway (it's not as pretty as expected, though).